components/logins/src/error.rs

```diff
@@ -79,3 +79,29 @@ pub enum InvalidLogin {
     #[fail(display = "Login has illegal field: {}", _0)]
     IllegalFieldValue { field_info: String },
 }
+
+impl Error {
+    // Get a short textual label identifying the type of error that occurred,
+    // but without including any potentially-sensitive information.
+    pub fn label(&self) -> &'static str {
```

Comment:
I don't like all this boilerplate, but AFAICT there's no easy way to get this sort of thing for free.

I also note that this overlaps somewhat with the way we map errors onto numeric codes for communicating them across the FFI. I'm open to the suggestion that we should return said numeric codes rather than strings, but the strings seem more useful for humans to deal with.

Reply:
It's probably fine, we'll get an error if an addition is made and this doesn't get updated.

You could probably get it for free out of the Debug implementation and some string processing, but this is cleaner.